pkg/controller/installation/core_controller.go

```diff
@@ -1179,15 +1171,17 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 			typhaNodeTLS.TrustedBundle.AddCertificates(prometheusClientCert)
 		}
 
-		// The following certificate is mounted by the es-kube-controllers for accessing Elasticsearch (Gateway).
+		// es-kube-controllers needs to trust the ESGW certificate. We'll fetch it here and add it to the trusted bundle.
+		// Note that although we're adding this to the typhaNodeTLS trusted bundle, it will be used by es-kube-controllers. This is because
+		// all components within this namespace share a trusted CA bundle.
 		esgwCertificate, err := certificateManager.GetCertificate(r.client, relasticsearch.PublicCertSecret, common.OperatorNamespace())
 		if err != nil {
 			r.status.SetDegraded(operator.CertificateError, fmt.Sprintf("Failed to retrieve / validate  %s", relasticsearch.PublicCertSecret), err, reqLogger)
 			return reconcile.Result{}, err
-		}
-		if esgwCertificate != nil {
+		} else if esgwCertificate != nil {
```

Comment:
I guess this is a difference of style, but I've learnt that else should not be used if the if-case returns/breaks.